test/test_utils.jl

```diff
@@ -89,3 +89,9 @@ end
     @test Fruits.APPLE isa Fruits.Fruit
     @test Fruits.ORANGE isa Fruits.Fruit
 end
+
+@testset "Test undef component prints" begin
```

Comment:
maybe do a test with both #undef and defined value to increase coverage? not sure if it's needed